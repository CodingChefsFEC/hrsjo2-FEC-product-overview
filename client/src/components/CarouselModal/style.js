import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';
const superLightGray = 'rgb(244, 244, 244)'

export const Container = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  height: 100vh;
  top: 0;
  left: 0;
  background-color: white;

  .topRow {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: 0.5rem 4rem;
    height: 88%;
  }

  .fa-times-circle {
    position: fixed;
    top: 3rem;
    right: 4rem;
  }
  .fa-times-circle:hover {
    cursor: pointer;
  }
`;

export const Thumbnails = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  margin-bottom: 0.5rem;
`;

export const Thumbnail = styled.div`
    margin: 0 5px;
    border: ${props => props.selected ? '2px' : '1px'} solid ${lightGray};
    overflow: hidden;
    height: 7.7vh;
    width: 5%;
    opacity: ${props => props.selected ? '1' : '0.25'};
    cursor: ${props => props.selected ? 'default' : 'pointer'};

    background-image: url("${props => props.image}");
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center;
    background-color: ${props => props.image ? 'transparent' : superLightGray};

    display: flex;
    justify-content: center;
    align-items: center;

    .share-pic-text {
      font-size: 8px;
      font-weight: bold;
      color: black;
    }

`;

export const MainImage = styled.div`
  height: 100%;
  min-width: 40%;
  max-width: 80%;

  background-image: url("${props => props.image}");
  background-color: ${props => props.image ? 'transparent' : superLightGray};
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
`;

export const PhotoUpload = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;

  .photo-upload--text {
    margin: 0 10rem;
  }
  .photo-upload--button {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 1rem;
    background-color: white;
    padding: 0.6rem;
    border: 1px solid ${lightGray};
    border-radius: 3px;
    cursor: pointer;
  }
  .photo-upload--button:hover {
    background-color: ${superLightGray};
  }
  .camera-icon {
    margin-right: .6rem;
  }
`;