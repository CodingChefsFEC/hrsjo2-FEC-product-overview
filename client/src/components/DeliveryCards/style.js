import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';
const superLightGray = 'rgb(244, 244, 244)';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  min-width: 22%;
  max-width: 25%;
  margin-top: 2rem;
`;

export const DeliveryCard = styled.div`
  display: flex;
  flex-direction: column;
  padding: .75rem 1rem;

  margin-bottom: 0.5rem;

  width: 100%;
  max-height: 30%;
  background-color: ${superLightGray};

  .top-half {
    margin-bottom: 0.5rem;
    height: 50%;
  }

  .green {
    color: ${darkGreen};
  }

  .local-area,
  .edit-location {
    text-decoration: underline;
  }

  .title {
    align-items: flex-end;
  }

  .at-to {
    margin: 0 0.25rem;
  }
  .edit-location {
    margin-top: 0.5rem;
  }

  .main-text {
    font-size: 13px;
    margin-bottom: 0.7rem;
  }
  .edit-location,
  .sub-text {
    font-size: 11px;
  }
`;

export const Row = styled.div`
  display: flex;
  flex-wrap: wrap;

  .option,
  .local-area {
    font-weight: bold;
  }

  .left {
    max-width: 50%;
  }
`;

export const Col = styled.div`
  display: flex;
  flex-direction: column;
  height: 50%;

  .title {

  }
`;

export const Button = styled.button`
  border: none;
  border-radius: 3px;
  color: white;
  font-size: 14px;

  height: 2rem;
  width: 45%;
  background-color: ${targetRed};

  margin-left: 0.75rem;
`;

export const Title = styled(Row)`
`;