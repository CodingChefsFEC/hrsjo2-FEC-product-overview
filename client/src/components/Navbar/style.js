import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';

export const Container = styled.div`
  display: flex;
  justify-content: space-between;

  width: 100%;
  height: 70px;

  color: white;
  background-color: ${targetRed};

  .nav-links {
    margin-right: 1rem;
    padding-top: 3px;
  }


  .icon {
    fill: white;
  }
  .icon:hover {
    cursor: pointer;
  }

  .logo {
    margin: 0.8rem 1rem;
  }

  .sign-in {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  .sign-in:hover {
    cursor: pointer;
  }

  .left-group,
  .right-group {
    display: flex;
    align-items: center;
  }
  .left-group {
    font-weight: bold;
    font-size: 16px;
  }
  .right-group {
    font-size: 12px;
  }
  .right {
    margin-right: 1.5rem;
  }
`;

export const NavLinks = styled.div`
  margin-right: 1rem;
  padding-top: 3px;

  &:hover {
    text-decoration: underline;
    cursor: pointer;
  }

  .fa-chevron-down {
    font-size: 12px;
    margin-left: 0.5rem;
  }
`;

export const SearchBar = styled.div`
  position: relative;

  .search-bar--input {
    width: 30rem;
    border: none;
    border-radius: 30px;
    height: 2.2rem;
    padding: 10px 72px 10px 16px;
    color: ${darkGray};
  }

  .search-bar--button {
    position: absolute;
    right: 0;
    border: none;
    border-radius: 0 30px 30px 0;
    height: 2.2rem;
    padding: 10px 12px 10px 10px;
  }
`;