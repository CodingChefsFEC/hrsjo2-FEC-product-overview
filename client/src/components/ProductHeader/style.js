import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';

export const Breadcrumbs = styled.div`
    color: ${lightGray};
    font-size: 12px;
    padding: 4px 4px;

    .divider {
        margin: 0 0.2rem;
    }
`;

export const ProductName = styled.h1`
    font-size: 23px;
    line-height: 1.25;
    margin-top: 4px;
    margin-bottom: 4px;
`;