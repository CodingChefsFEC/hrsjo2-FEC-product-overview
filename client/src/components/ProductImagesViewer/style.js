import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';

export const Container = styled.div`
    display: flex;
    /* justify-content: center; */
    border: 1px solid red;
    width: 60%;
`;

export const Thumbnails = styled.div`
    display: flex;
    flex-direction: column;
    margin-right: 6px;
    /* padding-left: 12px; */
`;

export const Thumbnail = styled.div`
    margin: 3px 2px 2px 0px;
    border: 1px solid ${lightGray};
    height: 100px;
    width: 100px;

    background-image: url("${props => props.imageURL}");
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center;

    &:hover {
        cursor: pointer;
    }

    .shadowOverlay {
        display: flex;
        align-items: center;
        justify-content: center;
        height: 100%;
        width: 100%;
        background-color: rgba(0, 0, 0, 0.6);
        color: white;
    }
`;

export const DarkWrapper = styled.div`
    background-color: black;
    opacity: 0.4;
    color: white;
    align-items: center;
    justify-content: center;
`;

export const MainImageContainer = styled.div`
    margin: 3px 2px 2px 1px;
    width: 100%;
    height: 100%;
    position: relative;

    overflow: hidden;
    &:hover {
        border: 1px solid ${darkGray};
    }
`;

export const ZoomLens = styled.div`
    position: absolute;
    /* border: 1px solid #d4d4d4; */
    width: 100px;
    height: 100px;
    &:hover {
        cursor: crosshair;
    }
`;

export const MainImage = styled.img`
    position: absolute;
    top: 0;
    left: 0;
    object-fit: contain;
    width: 525px;
    height: 525px;

    &:hover {
        cursor: crosshair;
        opacity: 0;
    }
`;

export const ZoomedImage = styled.div`
    width: 525px;
    height: 525px;
    background-image: url(${props => props.imageUrl});


    &:hover {
        cursor: crosshair;
        opacity: 1;
    }
`;