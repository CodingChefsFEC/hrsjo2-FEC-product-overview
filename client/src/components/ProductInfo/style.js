import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';
const starYellow = 'rgb(255, 215, 0)';

export const ProductInfoContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 33.33%;
`;

export const Container = styled.div`
  display: flex;
`;

export const Button = styled.a`
  font-size: 12px;
`;

export const Price = styled.h1`
  ${props => props.sale && `color: ${targetRed}`}
`;

export const SaleText = styled.span`
  color: ${targetRed};

  .sale {
    font-size: 23px;
  }
  .date {
    font-size: 14px;
  }
`;

export const Savings = styled.div`
  font-size: 12px;
  color: ${lightGray};
`;

export const Ratings = styled.div`
  display: flex;
  font-size: 12px;
  color: ${lightGray};

  .total-ratings {
    text-decoration: underline;
    color: ${lightGray};
  }
  .total-ratings:hover {
    text-decoration: none;
  }

  .ratings-button {
    position: relative;
    display: flex;
    background-color: white;
    border: none;
    cursor: pointer;
  }

  .icons {
    position: absolute;
  }
  .fa-star {
    color: ${starYellow};
  }

  i {
    display: flex;
    align-items: center;
    vertical-align: middle;
  }

  .fa-stack {
    font-size: 0.75em;
  }

  .yellowBar {
    position: absolute;
    width: 75%;
    height: 1.5rem;
    background-color: ${starYellow};
    z-index: 0;
  }
`;

export const Stars = styled.div`
    display: flex;
    background: linear-gradient(90deg, ${starYellow} ${props => props.starPercent}%, gray ${props => props.starPercent}%);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    font-size: 14px;
`;

export const QuantitySelect = styled.select`
  &:hover {
    background: lightGray;
  }
`;