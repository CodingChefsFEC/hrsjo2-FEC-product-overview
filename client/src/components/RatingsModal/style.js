import styled from "styled-components";

const targetRed = 'rgb(204, 0, 0)';
const lightGray = 'rgb(102, 102, 102)';
const darkGray = 'rgb(51, 51, 51)';
const darkGreen = 'rgb(0, 102, 1)';
const warningOrange = 'rgb(184, 83, 0)';

export const Container = styled.div`
  position: absolute;
  z-index: 100;
  display: flex;
  flex-direction: column;
  /* justify-content: center; */
  top: 25px;
  left: -100%;

  overflow: visible;
  padding: 1rem 0.75rem;

  border: 1px solid ${darkGray};
  border-radius: 3px;
  background-color: white;
  color: ${lightGray};

  width: ${props => props.width}px;
  height: ${props => props.height}px;

  .header {
    margin-bottom: 0.5em;
  }

  .reviews-link {
    margin-top: 2em;
  }

  .pointer {
    position: absolute;
    top: -1.6rem;
    left: 0;
    right: 0;
    margin: 0 auto;
    border-top: 1px solid ${darkGray};
    border-left: 1px solid ${darkGray};
    height: 1rem;
    width: 1rem;
    background-color: white;
    transform: rotate(45deg);
  }
`;

export const Row = styled.div`
  position: relative;
  display: flex;
  width: 100%;
  margin-bottom: 2em;

  .avg-rating {
    margin-right: 0.5rem;
  }

`;

export const Col = styled.div`
  display: flex;
  flex-direction: column;
  width: ${props => props.width};
`;

export const StarBar = styled(Row)`
  width: ${props => props.width}px;
  align-items: center;
  margin-bottom: 0.2rem;

  span:hover {
    text-decoration: underline;
  }

  .left,
  .right {
    text-align: left;
  }

  .left {
    width: 12%;
  }

  .gray-bar {
    background-color: rgb(247, 247, 247);
    width: 70%;
    height: 6px;
    border-radius: 20px;
    margin: 0 1rem 0 0;
  }
  .green-bar {
    background-color: ${darkGreen};
    width: ${props => props.percent}%;
    height: 100%;
    border-radius: 20px;
  }
`;

