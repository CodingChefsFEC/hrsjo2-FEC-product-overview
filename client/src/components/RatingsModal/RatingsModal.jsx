import React, {useState, useEffect, useRef} from 'react';
import { Container, Row, Col, StarBar } from './style.js';
import { Stars } from '../ProductInfo/style.js';

const RatingsModal = ({totalRatings, avgRating, stars, starPercent, setShowRatingsModal}) => {
  const ref = useRef(null);
  const handleOutsideClick = (e) => {
    const {className} = e.target

    if (ref.current && !ref.current.contains(e.target) &&
      !className.includes('ratings') &&
      !className.includes('star') &&
      !className.includes('chevron')
    ) {
      setShowRatingsModal(false);
    }
  }

  useEffect(() => {
    // Bind the event listener
    document.addEventListener("mousedown", handleOutsideClick);
    return () => {
        // Unbind the event listener on clean up
        document.removeEventListener("mousedown", handleOutsideClick);
    };
  }, [ref]);

  return (
    <Container width={375} ref={ref}>
      <Row className="header">
        <h2 className="avg-rating">{`${avgRating} out of 5`}</h2>
        <Stars starPercent={starPercent}>
          <i className="fas fa-star"></i>
          <i className="fas fa-star"></i>
          <i className="fas fa-star"></i>
          <i className="fas fa-star"></i>
          <i className="fas fa-star"></i>
        </Stars>
        <div className="pointer"></div>
      </Row>
      <Row>
        <span>{`${totalRatings} guest ratings`}</span>
      </Row>

      <Col className="star-bars">
        {stars.map((star, i) => {
          const percent = Math.round((star / totalRatings) * 100);
          return (
            <StarBar width={375} percent={percent}>
              <span href="#" className="left">{`${5 - i} star${i !== 4 ? 's' : ''}`}</span>
              <span href="#" className="gray-bar">
                <div className="green-bar" percent={percent}></div>
              </span>
              <span href="#" className="right">{`${percent}%`}</span>
            </StarBar>
          );
        })}
      </Col>
      <a href="#" className="reviews-link">Show all reviews</a>

    </Container>
  )
};

export default RatingsModal;