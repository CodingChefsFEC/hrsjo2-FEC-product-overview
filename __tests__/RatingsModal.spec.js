import 'jsdom-global/register';
import React from 'react';
import { mount, shallow, render } from 'enzyme';
import RatingsModal from '../client/src/components/RatingsModal/RatingsModal.jsx';
import { Container, Row, Col, StarBar } from '../client/src/components/RatingsModal/style.js';
import { Stars } from '../client/src/components/ProductInfo/style.js';

describe('RatingsModal', () => {

  it('should render the star-bars correctly with a given percent', () => {
    const percent = 50;
    const component = render(<StarBar width={375} percent={percent} />);

    expect(component).toMatchSnapshot();
  });

  it('should render the Stars component correctly given a starPercent ', () => {
    const starPercent = 25;
    const component = render(<Stars starPercent={starPercent} />);

    expect(component).toMatchSnapshot();
  });

});