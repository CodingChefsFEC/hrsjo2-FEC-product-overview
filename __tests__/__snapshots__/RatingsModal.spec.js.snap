// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsModal should render the Stars component correctly given a starPercent  1`] = `
<div
  class="style__Stars-sc-1d1ag70-8 jIljdQ"
/>
`;

exports[`RatingsModal should render the star-bars correctly with a given percent 1`] = `
<div
  class="style__Row-e8eyjc-1 style__StarBar-e8eyjc-3 jnTOSY buQbmk"
  width="375"
/>
`;
